http-server/src/server.rs

```diff
@@ -609,7 +761,7 @@ struct ProcessValidatedRequest<L: Logger> {
 /// Process a verified request, it implies a POST request with content type JSON.
 async fn process_validated_request<L: Logger>(
 	input: ProcessValidatedRequest<L>,
-) -> Result<hyper::Response<hyper::Body>, HyperError> {
+) -> Result<hyper::Response<hyper::Body>, Infallible> {
```

Comment:
Is there a reason to return a `Result<_, Infallible>` at all, and not just scrap the `Result` entirely?

Reply:
I think the hyper service fn requires `Result` but I could be wrong.

Reply:
If that's true it may still be cleaner to pass `Ok(result)` or something than have the "wrong" type signature on the whole fn? But shrug :)

Reply:
Fixed